### PR TITLE
support for db schedule-based static role rotations

### PR DESCRIPTION
**Pull request by fairclothjm**
Add support for the upcoming 1.15 Vault feature
- https://github.com/hashicorp/vault/pull/22484

Related docs PR for the feature
- https://github.com/hashicorp/vault/pull/22863

### CODE REVIEW

File: api/v1beta1/vaultdynamicsecret_types.go [hunk 1]
```diff
@@ -104,6 +104,11 @@ type VaultStaticCredsMetaData struct {
 	// "time to live". This value is compared to the LastVaultRotation to
 	// determine if a password needs to be rotated
 	RotationPeriod int64 `json:"rotationPeriod"`
+	// RotationSchedule is a "chron style" string representing the allowed
```
Comment (swenson) [inline api/v1beta1/vaultdynamicsecret_types.go:107]:
```suggestion
	// RotationSchedule is a "cron style" string representing the allowed
```

File: chart/crds/secrets.hashicorp.com_vaultdynamicsecrets.yaml [hunk 2]
```diff
@@ -226,13 +226,19 @@ spec:
                       be rotated
                     format: int64
                     type: integer
+                  rotationSchedule:
+                    description: RotationSchedule is a "chron style" string representing
```
Comment (swenson) [inline chart/crds/secrets.hashicorp.com_vaultdynamicsecrets.yaml:230]:
```suggestion
                    description: RotationSchedule is a "cron style" string representing
```

File: config/crd/bases/secrets.hashicorp.com_vaultdynamicsecrets.yaml [hunk 3]
```diff
@@ -226,13 +226,19 @@ spec:
                       be rotated
                     format: int64
                     type: integer
+                  rotationSchedule:
+                    description: RotationSchedule is a "chron style" string representing
```
Comment (swenson) [inline config/crd/bases/secrets.hashicorp.com_vaultdynamicsecrets.yaml:230]:
```suggestion
                    description: RotationSchedule is a "cron style" string representing
```

File: docs/api/api-reference.md [hunk 4]
```diff
@@ -543,6 +543,7 @@ _Appears in:_
 | --- | --- |
 | `lastVaultRotation` _integer_ | LastVaultRotation represents the last time Vault rotated the password |
 | `rotationPeriod` _integer_ | RotationPeriod is number in seconds between each rotation, effectively a "time to live". This value is compared to the LastVaultRotation to determine if a password needs to be rotated |
+| `rotationSchedule` _string_ | RotationSchedule is a "chron style" string representing the allowed schedule for each rotation. e.g. "1 0 * * *" would rotate at one minute past midnight (00:01) every day. |
```
Comment (swenson) [inline docs/api/api-reference.md:546]:
```suggestion
| `rotationSchedule` _string_ | RotationSchedule is a "cron style" string representing the allowed schedule for each rotation. e.g. "1 0 * * *" would rotate at one minute past midnight (00:01) every day. |
```

File: api/v1beta1/vaultdynamicsecret_types.go (hunk 1 above)
Comment (fairclothjm) [reply api/v1beta1/vaultdynamicsecret_types.go:107]: ha, woops!

**Comment by fairclothjm** [comment]
Closing in favor of https://github.com/hashicorp/vault-secrets-operator/pull/369 so that we can use a Jira-named branch and revert the integration test changes until we have TFVP support. We will incorporate the suggestions by @swenson in the new PR. Thanks!

**Pull request closed without merging**